### DISCUSSION TITLE
Google Sheet Data Into Budibase

### DISCUSSION THREAD

**Discussion by CodyReeves** [Budibase/budibase#1890] [Show and tell]
Here the steps to use Google Sheet Data into Budibase!

1. Go to Google Sheets and publish the sheet to the web, make sure the sheet is publically accessible. (File > Publish To Web) 
2. In Budibase, create a new external source via the Rest API.
3. Use the domain as the `Url` and with the page path as a template for a GET Request
```
https://spreadsheets.google.com/feeds/cells/GoolgeSheetID/SheetPageNumber/public/full?alt=json
```
4. Run Query! 
5. Data Should return if done correctly
6. Setup your schema

Overall it took a big of data wrangling to get it working but worth it for easy quick access to a sheet from Budibase.

**Comment by shogunpurple** [comment]
@CodyReeves this is amazing! We didn't even know you could do this with budibase, and we built it!

A proper google sheets integration is on our roadmap but we aren't too sure when it will be prioritised and delivered. This is an excellent solution in the meantime. Thanks again :)

**Comment by Manishkparmar80** [comment]
@codyreeves  & @shogunpurple it's really amazing.

I am trying to achieve revese functionality, Exporting app data to local excel file. Any idea how to do it in budibase.

**Reply by joebudi** [reply]
Hey! Could you explain the entire process of what you would like to achieve? Is it a one-off export? A scheduled export? What data would you like to export - data from the Budibase internal dbs? Data from user tables?

**Reply by Manishkparmar80** [reply]
@joebudi  hi,  
I am using external DBs ( MS SQL DB). 
pls. refer below screenshot for more detail of requirement. 
![Export to Excel](https://user-images.githubusercontent.com/84949590/124153579-33060b00-dab2-11eb-89cd-6ebad6402bb3.JPG)

**Reply by Manishkparmar80** [reply]
@joebudi  also discussed under #1885

**Comment by incposecretariate** [comment]
I have a drive pdf url in my GoogleSheet. How do i display it in my Budibase app? Either modal or any other way.
